server/controllers/stock/index.js

```diff
@@ -329,21 +336,73 @@ function listLotsOrigins(req, res, next) {
  * create a new integration entry
  */
 function createIntegration(req, res, next) {
+  const transaction = db.transaction();
   const params = req.body;
   const identifier = uuid.v4();
+  const documentUuid = uuid.v4();
+  let commonInfos;
+
   const integration = {
     uuid        : db.bid(identifier),
     project_id  : req.session.project.id,
-    description : params.description || 'INTEGRATION',
-    date        : new Date(),
+    description : params.integration.description || 'INTEGRATION',
+    date        : new Date(params.integration.date),
   };
-  const sql = `
-    INSERT INTO integration SET ?
-  `;
-  db.exec(sql, [integration])
-  .then(() => res.status(200).json(identifier))
-  .catch(next)
-  .done();
+  const sql = `INSERT INTO integration SET ?`;
+
+  transaction.addQuery(sql, [integration]);
+
+  params.lots.forEach((lot) => {
+    let lotUuid = uuid.v4();
+
+    // adding a lot insertion query into the transaction
+    transaction.addQuery(`INSERT INTO lot SET ?`, {
+      uuid             : db.bid(lotUuid),
+      label            : lot.label,
+      initial_quantity : lot.quantity,
+      quantity         : lot.quantity,
+      unit_cost        : lot.unit_cost,
+      expiration_date  : new Date(lot.expiration_date),
+      inventory_uuid   : db.bid(lot.inventory_uuid),
+      origin_uuid      : db.bid(identifier),
+      delay            : 0,
+    });
+
+    // adding a movement insertion query into the transaction
+    transaction.addQuery(`INSERT INTO stock_movement SET ?`, {
+      uuid          : db.bid(uuid.v4()),
+      lot_uuid      : db.bid(lotUuid),
+      depot_uuid    : db.bid(params.movement.depot_uuid),
+      document_uuid : db.bid(documentUuid),
+      flux_id       : params.movement.flux_id,
+      date          : new Date(params.movement.date),
+      quantity      : lot.quantity,
+      unit_cost     : lot.unit_cost,
+      is_exit       : 0,
+      user_id       : params.movement.user_id,
+    });
+
+    // An arry of common info, to send to the store procedure in order to insert to the posting journal
+    commonInfos = [ 
+      db.bid(documentUuid),
+      new Date(params.movement.date),
+      req.session.enterprise.id,
+      req.session.project.id,
+      req.session.enterprise.currency_id,
+      req.session.user.id
+    ];
+
+    // writting all records relative to the movement in the posting journal table
+    transaction.addQuery('CALL PostIntegration(?)', [commonInfos]);
```

Comment:
Does this produce any kind of financial document a user can print out?  It looks like it just writes straight into the Posting Journal.  Should it link to some form of document somewhere?  Could it create an intermediary voucher?

Reply:
There is a document produced if the integration is done with success but I can not say if it is a financial document or not.

Reply:
From the code, it looks like this may suffer from the same bug as #2102.  Can you verify that multiple lots will not result in multiple transactions?

Reply:
Oops! you are right, fixed now!